src/render.jl

```diff
@@ -34,29 +44,60 @@ type Config
         for (k, v) in merge(defaults, Dict(args))
             try
                 k in fields ? setfield!(this, k, v) : warn("Invalid setting: '$(k) = $(v)'.")
-            # e.g. catch TypeError
-            catch err
+            catch err # e.g. TypeError
                 warn("Invalid setting: '$(k) = $(v)'. Error: $err")
             end
         end
-        # Validations
-        for k in [:mdstyle_header, :mdstyle_objname, :mdstyle_meta, :mdstyle_exported, :mdstyle_internal]
-           getfield(this, k) in vcat(MDHTAGS, MDSTYLETAGS) ||
-                                error("""Invalid mdstyle value: config-item `$k -> $(getfield(this, k))`.
-                                        Valid values: [$(join(vcat(MDHTAGS, MDSTYLETAGS), ", "))].""")
+
+        for k in [:mdstyle_header, :mdstyle_objname, :mdstyle_meta, :mdstyle_subheader,
+                                                                    :mdstyle_genindex_mod]
+            getfield(this, k) in vcat(MDHTAGS, MDSTYLETAGS) ||
+                    error("""Invalid mdstyle : config-item `$k -> $(getfield(this, k))`.
+                          Valid values: [$(join(vcat(MDHTAGS, MDSTYLETAGS), ", "))].""")
         end
+
+        getfield(this, :md_subheader) in MD_SUBHEADER_OPTIONS ||
+                    error("""Invalid md_subheader : config-item `$k -> $(getfield(this, k))`.
+                          Valid values: $MD_SUBHEADER_OPTIONS.""")
         return this
     end
 end
 
+function update_config(config::Config; args...)
+    const fields   = fieldnames(Config)
+    for (k, v) in Dict(args)
+        try
+            k in fields ? setfield!(config, k, v) : warn("Invalid setting: '$(k) = $(v)'.")
+        catch err # e.g. TypeError
+            warn("Invalid setting: '$(k) = $(v)'. Error: $err")
+        end
+    end
+
+    for k in [:mdstyle_header, :mdstyle_objname, :mdstyle_meta, :mdstyle_subheader,
+                                                                :mdstyle_genindex_mod]
+        getfield(config, k) in vcat(MDHTAGS, MDSTYLETAGS) ||
+                error("""Invalid mdstyle : config-item `$k -> $(getfield(config, k))`.
+                      Valid values: [$(join(vcat(MDHTAGS, MDSTYLETAGS), ", "))].""")
+
+        getfield(config, :md_subheader) in MD_SUBHEADER_OPTIONS ||
+                    error("""Invalid md_subheader : config-item `$k -> $(getfield(config, k))`.
+                          Valid values: $MD_SUBHEADER_OPTIONS.""")
+    end
+    return config
+end
```

Comment:
Could the duplicate error checking in this method and the constructor be pulled into a separate helper function?